File: cosmos-master/code/sorting/bubble_sort/bubble_sort.swift
#!/usr/bin/env swift

/* Part of Cosmos by OpenGenus Foundation */
import Foundation

func sort(_ list: [Int]) -> [Int] {
    let sortedList = NSMutableArray(array: list)
    var sortedAboveIndex = list.count
    var swaps = 0

    repeat {
        var lastSwapIndex = 0
        
        for i in 1..<sortedAboveIndex {
            if (sortedList[i - 1] as! Int) > (sortedList[i] as! Int) {
                sortedList.exchangeObject(at: i, withObjectAt: i-1)
                lastSwapIndex = i
                swaps += 1
            }
        }

        sortedAboveIndex = lastSwapIndex
    } while (sortedAboveIndex != 0)

    return sortedList as! [Int]
}

let list = [1, 6, 2, 4, 9, 0, 5, 3, 7, 8]
print("Before: ", list)
let sortedList = sort(list)
print("After: ", sortedList)

File: cosmos-master/code/sorting/bubble_sort/bubble_sort.cpp
/* Part of Cosmos by OpenGenus Foundation */
#include <iostream>
#include <vector>
#include <algorithm>

void bubble_sort(std::vector<int> &v)
{
	for(int i=0; i < v.size(); ++i)
		for(int j=i+1; j < v.size(); ++j)
		{
			if (v[i] > v[j])
				std::swap(v[i],v[j]);
		}
}
int main()
{
	std::vector<int> input_array;
	input_array.push_back(2);
	input_array.push_back(7);
	input_array.push_back(1);
	input_array.push_back(10);
	input_array.push_back(-3);
	input_array.push_back(8);
	bubble_sort(input_array);
	for(auto x :input_array)
		std::cout<<x<<" ";
	std::cout<<"\n";
	return 0;
}
